### PR TITLE
Fix DataLogging for all Live data Modes

### PR DESCRIPTION
**Pull request by retiutut**
- [x] Need help adding a method somewhere to consume `get_timestamp_channel()` from Brainflow boards #680 
- [x] Get timestamp and pass this to `saveDataToFile()` instead of `new Date().getTime()`
- [x] Ensure millisecond accuracy for timestamp saved to file
- [x] Check that timestamps progress logically (time --->)
- [x] Cyton Accel
- [ ] Cyton Analog Mode
- [ ] Cyton Digital Mode
- [x] Ganglion
- [x] Cyton+WiFi
- [x] Ganglion+WiFi
- [ ] Double check using Windows and Linux

### CODE REVIEW

File: OpenBCI_GUI/WidgetManager.pde [hunk 1]
```diff
@@ -47,12 +47,10 @@ void setupWidgets(PApplet _this, ArrayList<Widget> w){
...
-    if (currentBoard instanceof AccelerometerCapableBoard) {
-        //Widget_2
-        w_accelerometer = new W_Accelerometer(_this);
-        w_accelerometer.setTitle("Accelerometer");
-        addWidget(w_accelerometer, w);
-    }
+    //Widget_2
+    w_accelerometer = new W_Accelerometer(_this);
+    w_accelerometer.setTitle("Accelerometer");
+    addWidget(w_accelerometer, w);
```
Comment (daniellasry) [inline OpenBCI_GUI/WidgetManager.pde]: explain this?

File: OpenBCI_GUI/DataProcessing.pde [hunk 2]
```diff
@@ -84,31 +70,11 @@ int getDataIfAvailable(int pointCounter) {
...
-    } else if (eegDataSource == DATASOURCE_SYNTHETIC) {
-
-        while ( (curDataPacketInd != lastReadDataPacketInd) && (pointCounter < nPointsPerUpdate)) {
-            lastReadDataPacketInd = (lastReadDataPacketInd+1) % dataPacketBuff.length;  //increment to read the next packet
-            
-            for (int Ichan=0; Ichan < nchan; Ichan++) {   //loop over each cahnnel
-                //scale the data into engineering units ("microvolts") and save to the "little buffer"
-                yLittleBuff_uV[Ichan][pointCounter] = dataPacketBuff[lastReadDataPacketInd].values[Ichan] * scaler;
+            if (currentBoard instanceof BoardCyton ||
+                currentBoard instanceof BoardGanglion) {
```
Comment (daniellasry) [inline OpenBCI_GUI/DataProcessing.pde]: This should probably be if currentBoard instanceof AccelerometerCapableBoard

File: OpenBCI_GUI/DataProcessing.pde [hunk 3]
```diff
@@ -58,24 +58,10 @@ int getDataIfAvailable(int pointCounter) {
...
-            // todo[brainflow] This method will be used to save data to ODF or BDF playback file
-            //println(timestamp + " | " + pointCounter % (timestamps.length + 1) + " of " + timestamps.length);
-            //saveDataToFile(scaler, lastReadDataPacketInd, timestamp,  currentBoard.getLastValidAccelValues());
-            pointCounter++; //increment counter for "little buffer"
-        }
-    } else if (eegDataSource == DATASOURCE_GANGLION) {
+    if (currentBoard instanceof BoardCyton ||
+        currentBoard instanceof BoardGanglion ||
+        currentBoard instanceof BoardNovaXR ||
+        currentBoard instanceof BoardSynthetic) {
```
Comment (daniellasry) [inline OpenBCI_GUI/DataProcessing.pde]: Isn't this basically all boards?

Comment (retiutut) [reply OpenBCI_GUI/DataProcessing.pde]: Not playback

File: OpenBCI_GUI/DataProcessing.pde (hunk 2 above)
Comment (retiutut) [reply OpenBCI_GUI/DataProcessing.pde]: Only save data to file for Cyton and Ganglion

Comment (daniellasry) [reply OpenBCI_GUI/DataProcessing.pde]: Ah, i see

File: OpenBCI_GUI/DataProcessing.pde (hunk 3 above)
Comment (daniellasry) [reply OpenBCI_GUI/DataProcessing.pde]: Would it make more sense to say if (! currentboard instanceof BoardPlayback) ?

Comment (retiutut) [reply OpenBCI_GUI/DataProcessing.pde]: Then this code would be broken until BoardPlayback is merged.

File: OpenBCI_GUI/WidgetManager.pde (hunk 1 above)
Comment (retiutut) [reply OpenBCI_GUI/WidgetManager.pde]: Accelerometer Widget will be used for all data modes except BoardNovaXr.

File: OpenBCI_GUI/DataProcessing.pde (hunk 3 above)
Comment (daniellasry) [reply OpenBCI_GUI/DataProcessing.pde]: i see

**Pull request closed without merging**